drivers/upscode2.c

```diff
@@ -168,273 +168,273 @@ static int
+	{ "BO", t_alarm,  "Battery over voltage", 0, NULL, NULL },
+	{ "BP", t_alarm,  "Bypass fuse failure", 0, NULL, NULL },
+	{ "BR", t_alarm,  "Abnormal battery recharge", 0, NULL, NULL },
+	{ "BT", t_alarm,  "Battery over temperature", 0, NULL, NULL },
+	{ "BX", t_alarm,  "Bypass unavailable", 0, NULL, NULL },
+	{ "BY", t_alarm,  "Battery failure", 0, NULL, NULL },
+	{ "CE", t_alarm,  "Configuration error", 0, NULL, NULL },
+	{ "CM", t_alarm,  "Battery converter failure", 0, NULL, NULL },
+	{ "CT", t_alarm,  "Cabinet over temperature", 0, NULL, NULL },
+	{ "DO", t_alarm,  "DC over voltage", 0, NULL, NULL },
+	{ "DU", t_alarm,  "DC under voltage", 0, NULL, NULL },
+	{ "EP", t_alarm,  "Emergency power off", 0, NULL, NULL },
+	{ "FF", t_alarm,  "Fan failure", 0, NULL, NULL },
+	{ "FH", t_alarm,  "Line frequency high", 0, NULL, NULL },
+	{ "FL", t_alarm,  "Line frequency low", 0, NULL, NULL },
+	{ "FT", t_alarm,  "Filter over temperature", 0, NULL, NULL },
+	{ "GF", t_alarm,  "Ground failure", 0, NULL, NULL },
+	{ "HT", t_alarm,  "Charger over temperature", 0, NULL, NULL },
+	{ "IB", t_alarm,  "Internal data bus failure", 0, NULL, NULL },
+	{ "IF", t_alarm,  "Inverter fuse failure", 0, NULL, NULL },
+	{ "IM", t_alarm,  "Inverter failure", 0, NULL, NULL },
+	{ "IO", t_alarm,  "Inverter over voltage", 0, NULL, NULL },
+	{ "IP", t_alarm,  "Internal power supply failure", 0, NULL, NULL },
+	{ "IT", t_alarm,  "Inverter over temperature", 0, NULL, NULL },
+	{ "IU", t_alarm,  "Inverter under voltage", 0, NULL, NULL },
+	{ "IV", t_alarm,  "Inverter off", 0, NULL, NULL },
+	{ "LR", t_alarm,  "Loss of redundancy", 0, NULL, NULL },
+	{ "NF", t_alarm,  "Neutral fault", 0, NULL, NULL },
+	{ "OD", t_status, "UPS not supplying load", UPSC_STAT_OFF, NULL, NULL },
+	{ "OF", t_alarm,  "Oscillator failure", 0, NULL, NULL },
+	{ "OL", t_status, "Overload", UPSC_STAT_OVERLOAD, NULL, NULL },
+	{ "OR", t_alarm,  "Redundancy overload", 0, NULL, NULL },
+	{ "OV", t_alarm,  "Abnormal output voltage", 0, NULL, NULL },
+	{ "OW", t_alarm,  "Output failure", 0, NULL, NULL },
+	{ "PB", t_alarm,  "Parallel bus failure", 0, NULL, NULL },
+	{ "PE", t_alarm,  "Phase rotation error", 0, NULL, NULL },
+	{ "RE", t_alarm,  "Rectifier off", 0, NULL, NULL },
+	{ "RF", t_alarm,  "Rectifier fuse failure", 0, NULL, NULL },
+	{ "RM", t_alarm,  "Rectifier failure", 0, NULL, NULL },
+	{ "RT", t_alarm,  "Rectifier over temperature", 0, NULL, NULL },
+	{ "SM", t_alarm,  "Static switch failure", 0, NULL, NULL },
+	{ "ST", t_alarm,  "Static switch over temperature", 0, NULL, NULL },
+	{ "TT", t_alarm,  "Trafo over temperature", 0, NULL, NULL },
+	{ "UD", t_alarm,  "UPS disabled", 0, NULL, NULL },
+	{ "UO", t_alarm,  "Utility over voltage", 0, NULL, NULL },
+	{ "US", t_alarm,  "Unsynchronized", 0, NULL, NULL },
+	{ "UU", t_alarm,  "Utility under voltage", 0, NULL, NULL },
+	{ "VE", t_alarm,  "internal voltage error", 0, NULL, NULL },
+	{ NULL, t_ignore, NULL, 0, NULL, NULL }
```

Comment:
Point for discussion: not sure if `t_ignore` is a good sentinel value here and in other tables of this file. Not verified in practice.